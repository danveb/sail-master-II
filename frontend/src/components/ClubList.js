import React, { useState, useEffect } from 'react' 
import ClubCard from './ClubCard'
import SailMasterIIApi from '../API/api'
import { Container, Row, Col } from 'react-bootstrap'

const ClubList = () => {
    // useState 
    const [clubs, setClubs] = useState([])

    // useEffect
    useEffect(() => {
        async function getClubs() {
            const response = await SailMasterIIApi.getClubs() 
            setClubs(response)
        }
        getClubs() 
    }, [setClubs])

    return (
        <Container className="d-flex align-items-center justify-content-center mt-4"> 
            <Row>
                <h1 className="text-center mb-4">Sailing Clubs</h1> 
                { clubs.map((club) =>
                    <Col md={4} >
                        <ClubCard 
                            key={club.id}
                            id={club.id}
                            name={club.name}
                            city={club.city}
                            state={club.state}
                            zip={club.zip}
                            lat={club.lat}
                            lon={club.lon}
                        />
                    </Col>
                ) }
            </Row>
        </Container>
    )
}

export default ClubList