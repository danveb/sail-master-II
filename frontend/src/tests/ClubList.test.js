import React from 'react'
import { render } from '@testing-library/react'
import ClubList from '../components/ClubList'
// smoke test 
it('renders without crashing', () => {
    render(<ClubList />) 
})

// snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<ClubList />)
    expect(asFragment()).toMatchSnapshot()
})