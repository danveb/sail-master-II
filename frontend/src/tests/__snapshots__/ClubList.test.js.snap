// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center mt-4 container"
  >
    <div
      class="row"
    >
      <h1
        class="text-center mb-4"
      >
        Sailing Clubs
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
